vector-stores/spring-ai-gemfire/src/main/java/org/springframework/ai/vectorstore/GemFireVectorStore.java

```diff
@@ -279,57 +305,109 @@ public void add(List<Document> documents) {
-			.bodyValue(upload)
-			.retrieve()
-			.bodyToMono(Void.class)
-			.block();
+		ObjectMapper objectMapper = new ObjectMapper();
+		try {
+			String embeddingsJson = objectMapper.writeValueAsString(upload).substring("{\"embeddings\":".length());
+
+			client.post()
+				.uri("/" + INDEX_NAME + "/embeddings")
+				.contentType(MediaType.APPLICATION_JSON)
+				.bodyValue(embeddingsJson)
+				.retrieve()
+				.bodyToMono(Void.class)
+				.block();
+		}
+		catch (WebClientResponseException.NotFound notFoundException) {
+			throw new RuntimeException("Vector index does not exist", notFoundException);
+		}
+		catch (WebClientResponseException.BadRequest badRequestException) {
+			throw new RuntimeException("Bad request. The request body could not be processed successfully.",
+					badRequestException);
+		}
+		catch (Exception e) {
+			throw new RuntimeException("An unexpected error occurred", e);
+		}
 	}
 
 	@Override
 	public Optional<Boolean> delete(List<String> idList) {
-		throw new UnsupportedOperationException("This vector store doesn't support delete (yet)");
+		try {
+			client.method(HttpMethod.DELETE)
+				.uri("/" + INDEX_NAME + "/embeddings/")
+				.body(BodyInserters.fromValue(idList))
+				.retrieve()
+				.bodyToMono(Void.class)
+				.block();
+		}
+		catch (Exception e) {
+			throw new RuntimeException("Error removing embedding " + e);
+		}
+		return Optional.of(true);
 	}
 
 	@Override
 	public List<Document> similaritySearch(SearchRequest request) {
-		List<Double> vector = this.embeddingClient.embed(request.getQuery());
-
-		return client.post()
-			.uri("/query")
-			.contentType(MediaType.APPLICATION_JSON)
-			.bodyValue(new QueryRequest(vector, request.getTopK(), topKPerBucket, true))
-			.retrieve()
-			.bodyToFlux(QueryResponse.class)
-			.filter(r -> r.score >= request.getSimilarityThreshold())
-			.map(r -> {
-				Map<String, Object> metadata = r.metadata;
-				metadata.put(DISTANCE_METADATA_FIELD_NAME, 1 - r.score);
-				String content = (String) metadata.remove(documentField);
-
-				return new Document(r.key, content, metadata);
-			})
-			.collectList()
-			.block();
+		try {
+			List<Double> vector = this.embeddingClient.embed(request.getQuery());
+			List<Float> floatVector = vector.stream().map(Double::floatValue).toList();
+
+			return client.post()
+				.uri("/" + INDEX_NAME + "/query")
+				.contentType(MediaType.APPLICATION_JSON)
+				.bodyValue(new QueryRequest(floatVector, request.getTopK(), topKPerBucket, true))
+				.retrieve()
+				.bodyToFlux(QueryResponse.class)
+				.filter(r -> r.score >= request.getSimilarityThreshold())
+				.map(r -> {
+					Map<String, Object> metadata = r.metadata;
+					metadata.put(DISTANCE_METADATA_FIELD_NAME, 1 - r.score);
+					String content = (String) metadata.remove(documentField);
+
+					return new Document(r.key, content, metadata);
+				})
+				.collectList()
+				.block();
+		}
+		catch (Exception e) {
+			throw new RuntimeException("Error in similarity search", e);
```

Comment:
Same question for most of the error handling actually.  Is the spring ai api expecting these exceptions or are they hoping we handle them and return empty results/ false etc?